test/ignore.spec.js

```diff
@@ -0,0 +1,20 @@
+import Alpine from 'alpinejs'
+import { wait } from '@testing-library/dom'
+
+global.MutationObserver = class {
+    observe() {}
+}
+
+test('x-ignore ignores initialization of any new components within the node', async () => {
+    document.body.innerHTML = `
+    <div x-ignore>
```

Comment:
What happens if x-ignore is on the same element as x-data?

Reply:
To watch root ignore element by alpine,  it will be initialized as component. Because we might use it as :
```html
        <div x-bind:x-ignore="ignore" x-data="{ignore: true}">
            <div x-data="{foo:'foo'}">
                <span x-text="foo"></span>
            </div>
        </div>
``` 

so if ignore condition is true then it can be ignored. This is the logic for now. I hope it makes sense.

Reply:
Actually i think i need to adjust this part. Let me see.

Reply:
should be handled by ab545c15b0cf855ea93b29c9232d10d707a492a3